include/eve/arch/cpu/wide.hpp

```diff
@@ -991,6 +991,37 @@ namespace eve
   //! @}
   //================================================================================================
 
+  //====================================================================================================================
+  //! @name Deduction Guides
+  //! @{
+  //====================================================================================================================
+
+  /// Allows deduction from a single eve::scalar_value
+  template<scalar_value S> wide(S const&) -> wide<S, expected_cardinal_t<S>>;
+
+  /// Allows deduction from a @callable
+  template<eve::invocable<std::ptrdiff_t, std::ptrdiff_t> Generator>
+  wide(Generator&& g) -> wide<decltype( std::declval<Generator>()(0,0) )>;
```

Comment:
I think this one should be explicit tbh

Reply:
It's a CTAD, there is no notion of explicit.
THe constructor is though

Reply:
oh the ctor is not explicit indeed

Reply:
Well, writing

```c++
wide w = [](auto i, auto) {....];
```

is cool. Beign explicit will requries () around it.

Reply:
Sorry - I meant explicitly specifinh a type